### PR TITLE
TECH-317: Add infinite scroll to the Products table

### PR DESCRIPTION
**Pull request by KarolinaKopacz**
**Description:**
Add infinite scroll to the Products table.
Ticket: https://govstack-global.atlassian.net/browse/TECH-317

### CODE REVIEW

File: components/table/ProductTable.tsx [hunk 1]
```diff
@@ -32,10 +41,19 @@ const ProductTable = () => {
...
+        <div id='scrollableDiv' className='product-table-row-section'>
+          <InfiniteScroll
+            scrollableTarget='scrollableDiv'
+            dataLength={40}
+            next={handleLoadMoreData}
+            hasMore={hasMoreData}
+            loader={<p className='data-loader-message'>Loading more data...</p>}
+            style={{ overflowX: 'hidden' }}
+          >
+            {productsList.map((product, key) => (
```
Comment (dborowiecki) [inline components/table/ProductTable.tsx:53]: For SubTable we have SubTableRows and in here we're using iteration over productList to generate ProductTableRow, I think those two should be allined. Could you change ProductTableRow to ProductTableRows which will take productList as an input?

File: components/table/ProductTable.tsx [hunk 2]
```diff
@@ -32,10 +41,19 @@ const ProductTable = () => {
...
-          {productsList.map((product, key) => (
-            <ProductTableRow product={product} key={`product-${key}`} />
-          ))}
+        <div id='scrollableDiv' className='product-table-row-section'>
+          <InfiniteScroll
+            scrollableTarget='scrollableDiv'
+            dataLength={40}
+            next={handleLoadMoreData}
+            hasMore={hasMoreData}
+            loader={<p className='data-loader-message'>Loading more data...</p>}
```
Comment (dborowiecki) [inline components/table/ProductTable.tsx:50]: Can we put loader in separate component and reuse it across different parts of application? I'm not sure if at the moment that's only place where it's used, but in future we definitely have more scrolls and paginations, it be good to have consistent loading option. Also, I think `Loading more data...` should be translated.

File: components/table/ProductTable.tsx [hunk 3]
```diff
@@ -20,6 +23,12 @@ const ProductTable = () => {
 
   const productListLength = productsList.length;
 
+  const handleLoadMoreData = () => {
+    // mocked response
+    setProductsList([...productsList, ...moreMockedProductList]);
```
Comment (dborowiecki) [inline components/table/ProductTable.tsx:28]: When do we plan to attach backend data? Is [TECH-451](https://govstack-global.atlassian.net/browse/TECH-451) blocking it?

**Review by dborowiecki — CHANGES_REQUESTED** [review]

File: components/table/ProductTable.tsx (hunk 3 above)
Comment (KarolinaKopacz) [reply components/table/ProductTable.tsx:28]: [TECH-408](https://govstack-global.atlassian.net/browse/TECH-408) is blocking it.
I'm going to attach backend data while working on [TECH-318](https://govstack-global.atlassian.net/browse/TECH-318)

File: components/table/ProductTable.tsx (hunk 1 above)
Comment (KarolinaKopacz) [reply components/table/ProductTable.tsx:53]: Here is the difference. In ProductTableRow is actually one row, ProductTableRow component accept one product = one row, not list of products (SubTableRows accept list of building blocks).

File: components/table/ProductTable.tsx (hunk 2 above)
Comment (KarolinaKopacz) [reply components/table/ProductTable.tsx:50]: Good point!

File: components/table/ProductTable.tsx (hunk 1 above)
Comment (dborowiecki) [reply components/table/ProductTable.tsx:54]: Thanks, as we discussed, these components will be updated later, during work on TECH-318.

**Pull request merged**